pkg/network/usm/utils/connection_rollup_test.go

```diff
@@ -0,0 +1,136 @@
+
+//go:build linux_bpf
+
+package utils
+
+import (
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/network/types"
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRollupKey(t *testing.T) {
+	t.Run("same key", func(t *testing.T) {
+		aggregator := NewConnectionAggregator()
+		srcIP := util.AddressFromString("1.1.1.1")
+		dstIP := util.AddressFromString("2.2.2.2")
+
+		c1 := types.NewConnectionKey(srcIP, dstIP, 6000, 80)
+		t1 := aggregator.RollupKey(c1)
+		t2 := aggregator.RollupKey(c1)
+
+		assert.Equal(t, c1, t1)
+		assert.Equal(t, c1, t2)
+	})
+
+	t.Run("same key, flipped order", func(t *testing.T) {
+		aggregator := NewConnectionAggregator()
+		srcIP := util.AddressFromString("1.1.1.1")
+		dstIP := util.AddressFromString("2.2.2.2")
+
+		c1 := types.NewConnectionKey(srcIP, dstIP, 6000, 80)
+		t1 := aggregator.RollupKey(c1)
+
+		c2 := types.NewConnectionKey(dstIP, srcIP, 80, 6000)
+		t2 := aggregator.RollupKey(c2)
+
+		assert.Equal(t, c1, t1)
+		assert.Equal(t, c1, t2)
+	})
+
+	t.Run("same IPs, but no matching ports", func(t *testing.T) {
+		aggregator := NewConnectionAggregator()
+		srcIP := util.AddressFromString("1.1.1.1")
+		dstIP := util.AddressFromString("2.2.2.2")
+
+		c1 := types.NewConnectionKey(srcIP, dstIP, 6000, 80)
+		t1 := aggregator.RollupKey(c1)
+
+		c2 := types.NewConnectionKey(dstIP, srcIP, 7000, 53)
+		t2 := aggregator.RollupKey(c2)
+
+		// In this case both keys are preserved, which wouldn't trigger a rollup
+		assert.Equal(t, c1, t1)
+		assert.Equal(t, c2, t2)
+	})
+
+	t.Run("same IPs, different ephemeral ports", func(t *testing.T) {
+		aggregator := NewConnectionAggregator()
+		srcIP := util.AddressFromString("1.1.1.1")
+		dstIP := util.AddressFromString("2.2.2.2")
+
+		c1 := types.NewConnectionKey(srcIP, dstIP, 6000, 80)
+		t1 := aggregator.RollupKey(c1)
+
+		c2 := types.NewConnectionKey(srcIP, dstIP, 6001, 80)
+		t2 := aggregator.RollupKey(c2)
+
+		// Let's also try a different tuple order
+		c3 := types.NewConnectionKey(dstIP, srcIP, 80, 6002)
+		t3 := aggregator.RollupKey(c3)
+
+		// c1, c2 and c3 should all translate to c1
+		assert.Equal(t, c1, t1)
+		assert.Equal(t, c1, t2)
+		assert.Equal(t, c1, t3)
+	})
+}
+
+func TestClearEphemeralPort(t *testing.T) {
+	t.Run("no state", func(t *testing.T) {
+		aggregator := NewConnectionAggregator()
+		srcIP := util.AddressFromString("1.1.1.1")
+		dstIP := util.AddressFromString("2.2.2.2")
+
+		c1 := types.NewConnectionKey(srcIP, dstIP, 6000, 80)
+
+		// Nothing should happen in this case
+		assert.Equal(t, c1, aggregator.ClearEphemeralPort(c1))
+	})
+
+	t.Run("base case", func(t *testing.T) {
+		aggregator := NewConnectionAggregator()
+		srcIP := util.AddressFromString("1.1.1.1")
+		dstIP := util.AddressFromString("2.2.2.2")
+
+		c1 := types.NewConnectionKey(srcIP, dstIP, 6001, 80)
+		_ = aggregator.RollupKey(c1)
+		c2 := types.NewConnectionKey(srcIP, dstIP, 6002, 80)
+		_ = aggregator.RollupKey(c2)
```

Comment:
why do you call `RollupKey` without using its value?

Reply:
In this test scenario I'm mostly interested in the `ClearEphemeralPort` method (`RollupKey` Is being tested elsewhere)
The calls to `RollupKey` are basically only feeding state to the `ConnectionAggregator`